src/qutip_qip/circuit/mat_renderer.py

```diff
@@ -0,0 +1,1025 @@
+            fontfamily=self.fontfamily,
+            fontstyle=self.fontstyle,
+            verticalalignment="center",
+            horizontalalignment="center",
+            zorder=self._zorder["gate_label"],
+        )
+        gate_patch = FancyBboxPatch(
+            (
+                self._get_xskip([gate_wire], layer) + self.style.gate_margin,
+                (gate_wire + self._cwires) * self.style.wire_sep
+                - self._min_gate_height / 2,
+            ),
+            gate_width,
+            self._min_gate_height,
+            boxstyle=self.style.bulge,
+            mutation_scale=0.3,
+            facecolor=self.color,
+            edgecolor=self.color,
+            zorder=self._zorder["gate"],
+        )
+
+        self._ax.add_artist(gate_text)
+        self._ax.add_patch(gate_patch)
+        self._manage_layers(gate_width, [gate_wire], layer)
+
+    def _draw_multiq_gate(self, gate: Gate, layer: int) -> None:
+        """
+        Draw the multi-qubit gate.
+
+        Parameters
+        ----------
+        gate : Gate Object
+            The gate to be plotted.
+
+        layer : int
+            The layer the gate is acting on.
+        """
+
+        wire_list = list(
+            range(self.merged_qubits[0], self.merged_qubits[-1] + 1)
+        )
+        com_xskip = self._get_xskip(wire_list, layer)
+
+        if gate.name == "CNOT":
+            self._draw_control_node(gate.controls[0], com_xskip, self.color)
+            self._draw_target_node(gate.targets[0], com_xskip, self.color)
+            self._draw_qbridge(
+                gate.targets[0], gate.controls[0], com_xskip, self.color
+            )
+            self._manage_layers(
+                2 * self.style.gate_pad + self._target_node_r / 3,
+                wire_list,
+                layer,
+                com_xskip,
+            )
+
+        elif gate.name == "SWAP":
+            self._draw_swap_mark(gate.targets[0], com_xskip, self.color)
+            self._draw_swap_mark(gate.targets[1], com_xskip, self.color)
+            self._draw_qbridge(
+                gate.targets[0], gate.targets[1], com_xskip, self.color
+            )
+            self._manage_layers(
+                2 * (self.style.gate_pad + self._min_gate_width / 3),
+                wire_list,
+                layer,
+                com_xskip,
+            )
+
+        elif gate.name == "TOFFOLI":
+            self._draw_control_node(gate.controls[0], com_xskip, self.color)
+            self._draw_control_node(gate.controls[1], com_xskip, self.color)
+            self._draw_target_node(gate.targets[0], com_xskip, self.color)
+            self._draw_qbridge(
+                gate.targets[0], gate.controls[0], com_xskip, self.color
+            )
+            self._draw_qbridge(
+                gate.targets[0], gate.controls[1], com_xskip, self.color
+            )
+            self._manage_layers(
+                2 * self.style.gate_pad + self._target_node_r / 3,
```

Comment:
It seems that all the calling of _manage_layers takes `2 * self.style.gate_pad + self._target_node_r / 3`, Can we somehow define this as a separate function to reduce the duplication and make it easier to add new gates?
Something like a `_gate_pad_to_gate_width_map`?

Reply:
As of now only two calling take `2 * self.style.gate_pad + self._target_node_r / 3` as argument (CNOT and TOFFOLI). Calling inside `_draw_singleq_gate`, `SWAP` and  general `_draw_multiq_gate` are different.